### PR TITLE
DAOS-2902 Security: Fixes for Certificate Generation support

### PR DESCRIPTION
**Pull request by dpquigl**
Certificate generation and config examples needed some fixing to enable longer
certificate default lifetime. Also the rpm specs have been updated to package
these scripts and config files.

Signed-off-by: David Quigley <david.quigley@intel.com>

### CODE REVIEW

File: utils/rpms/daos.spec [hunk 1]
```diff
@@ -162,6 +163,12 @@ echo "%{_libdir}/daos_srv" > %{?buildroot}/%{_sysconfdir}/ld.so.conf.d/daos.conf
 %{_libdir}/libdaos_common.so
 # TODO: this should move to %{_libdir}/daos/libplacement.so
 %{_libdir}/daos_srv/libplacement.so
+# Certificate generation files
+%{daoshome}/certgen/admin.cnf
+%{daoshome}/certgen/agent.cnf
+%{daoshome}/certgen/ca.cnf
+%{daoshome}/certgen/server.cnf
+%{daoshome}/certgen/gen_certifictes.sh
```
Comment (brianjmurrell) [inline utils/rpms/daos.spec]: You can use:

```
%{daoshome}/certgen/
```

ere in place of all of those entries if you want to future-proof adding more files to this dir

File: utils/certs/SConscript [hunk 2]
```diff
@@ -0,0 +1,12 @@
+"""Build DAOS config"""
```
Comment (kjacque) [inline utils/certs/SConscript]: Change to "build DAOS certificate utilities" or something like that

File: utils/certs/ca.cnf [hunk 3]
```diff
@@ -12,7 +12,7 @@ certificate		= $dir/daosCA.crt
 private_key		= $dir/private/daosCA.key
 
 default_md		= sha512	# SAFE Crypto Requires SHA-512
-default_days		= 365		# how long to certify for
+default_days		= 3650		# how long to certify for
```
Comment (kjacque) [inline utils/certs/ca.cnf]: It looks like we pass in the days anyway in the script... Is this still useful?

File: utils/rpms/daos.spec [hunk 4]
```diff
@@ -128,7 +129,8 @@ PREFIX="%{?_prefix}"
 sed -i -e s/${BUILDROOT//\//\\/}[^\"]\*/${PREFIX//\//\\/}/g %{?buildroot}%{_prefix}/TESTING/.build_vars.*
 mv %{?buildroot}%{_prefix}/lib{,64}
 #mv %{?buildroot}/{usr/,}etc
-mkdir -p %{?buildroot}/%{daoshome}
+mkdir -p %{?buildroot}/%{_exec_prefix}/lib/%{name}
+mv %{?buildroot}%{_prefix}/lib64/daos %{?buildroot}/%{_exec_prefix}/lib/
```
Comment (brianjmurrell) [inline utils/rpms/daos.spec]: Hrm.  I'm surprised this works.  On line 132 you create `/usr/lib/daos/` and then on line 133 you move `/usr/lib64/daos` to `/usr/lib/`.  I'd expect you to get an error from the `mv`.  But it doesn't seem to.  It seems to happly replace an empty directory.  Learned something new.

**Pull request merged**